xarray/coding/cftime_offsets.py

```diff
@@ -122,6 +123,11 @@ def __sub__(self, other):
             return NotImplemented
 
     def __mul__(self, other):
+        if not isinstance(other, int):
+            raise TypeError(
+                "The provided 'other' must be an integer. "
+                "Instead a value of type {!r} was provided.".format(type(other))
+            )
```

Comment:
if I do not add this here, I get `unsupported operand type(s) for *: 'QuarterEnd' and 'float'"` for `Quarter` and `Year` but not `Month`

Reply:
Therefore I added this `TypeError` here to get consistent errors, but probably there is a better way

Reply:
using `return NotImplemented` here to get `unsupported operand type(s) for *`

Reply:
Perfect